standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -494,7 +494,7 @@ static void prewarm(RawStore rawStore) {
               AggrStats aggrStatsAllButDefaultPartition = null;
               if (!table.getPartitionKeys().isEmpty()) {
                 Deadline.startTimer("getPartitions");
-                partitions = rawStore.getPartitions(catName, dbName, tblName, Integer.MAX_VALUE);
+                partitions = rawStore.getPartitions(catName, dbName, tblName, -1);
```

Comment:
This comment is not for this patch. But I guess there are more users will experience this if they use limit=MAX_VALUE. Maybe we can ask Naveen to treat MAX_VALUE as -1.

Reply:
the limit = Integer.MAX_VALUE was used in a couple other places, e.g. in HiveMetaStore.java, I saw 
for (Partition partition : ms.getPartitions(catName, dbName, tableName, Integer.MAX_VALUE)) {
. Since RawStore is an interface, I am not sure how to change it to treat MAX_VALUE as -1. Also the exception was thrown from Mysql jdbc jar for value higher than 50million rows. The RawStore comment does indcate if no limit desired, we should pass in -1.  I am keep it here just for passing build/local test, etc. I guess we can fix other places in a separate jira.

Reply:
I have ping'ed Naveen to also take a look.